File: high_spirit_gym/lib/models/attendance.dart
class Attendance {
  final int attendanceID;
  final int customerID;
  final String customerName;
  final DateTime checkInTime;
  final DateTime? checkOutTime;
  final String? notes;

  Attendance({
    required this.attendanceID,
    required this.customerID,
    required this.customerName,
    required this.checkInTime,
    this.checkOutTime,
    this.notes,
  });

  bool get isCheckedIn => checkOutTime == null;

  String get durationText {
    if (checkOutTime == null) return 'Still in gym';
    final diff = checkOutTime!.difference(checkInTime);
    final hours = diff.inHours;
    final mins = diff.inMinutes % 60;
    if (hours > 0) return '${hours}h ${mins}m';
    return '${mins}m';
  }

  factory Attendance.fromJson(Map<String, dynamic> json) {
    return Attendance(
      attendanceID: json['attendanceID'] ?? 0,
      customerID: json['customerID'] ?? 0,
      customerName: json['customerName'] ?? '',
      checkInTime: DateTime.tryParse(json['checkInTime'] ?? '') ?? DateTime.now(),
      checkOutTime: json['checkOutTime'] != null ? DateTime.tryParse(json['checkOutTime']) : null,
      notes: json['notes'],
    );
  }
}

File: HighSpiritApp/Models/Attendance.cs
using System.ComponentModel.DataAnnotations;

namespace HighSpiritApp.Models
{
    public class Attendance
    {
        [Key]
        public int AttendanceID { get; set; }

        public int CustomerID { get; set; }

        public string CustomerName { get; set; } = string.Empty;

        public DateTime CheckInTime { get; set; } = DateTime.Now;

        public DateTime? CheckOutTime { get; set; }

        public string? Notes { get; set; }

        // Navigation
        public Customer? Customer { get; set; }
    }
}
